server/src/main/java/org/elasticsearch/common/Rounding.java

```diff
@@ -556,19 +567,73 @@ public boolean equals(Object obj) {
         }
     }
 
+    static class OffsetRounding extends Rounding {
+        static final byte ID = 3;
+
+        private final Rounding delegate;
+        private final long offset;
+
+        OffsetRounding(Rounding delegate, long offset) {
+            this.delegate = delegate;
+            this.offset = offset;
+        }
+
+        OffsetRounding(StreamInput in) throws IOException {
+            delegate = Rounding.read(in);
+            offset = in.readLong();
+        }
+
+        @Override
+        public void innerWriteTo(StreamOutput out) throws IOException {
+            if (out.getVersion().before(Version.V_8_0_0)) {
```

Comment:
We don't actually serialize this `Rounding` but I believe I've got this bit right so I kept it. I'd like to move the other `offset` code over to this Rounding which means that we will *eventually* serialize it. At that point we'll get exhaustive tests for this code.

Reply:
Modulo the version check which I discuss below.